content/docs/console/continuous-delivery/jenkins.md

```diff
@@ -0,0 +1,145 @@
+
+### Project Parameters (Environment Variables)
+
+In order to deploy to one of the cloud providers, you will need to ensure that the authentication environment variables are set, so that the Pulumi CLI can use them to deploy your infrastructure resources. The set of environment variables to configure vary for each cloud. For Azure, depending on your setup, you may have to set at most 4 environment variables. In this example, we will assume you are using a [Service Principal](https://www.pulumi.com/docs/reference/clouds/azure/setup/#creating-a-service-principal).
+
+The screenshot below shows you how you can parameterize your `Jenkinsfile` using environment variables.
+
+![Jenkins Project Parameters](/images/docs/reference/jenkins/project-params.png)
+
+#### Configuring the Node JS plugin
+
+In order to use the Node JS plugin, you must first ensure you add at least one installation to it. Navigate to the **Global Tool Configuration** menu option on the Manage Jenkins page.
+
+**Note** The name you enter in the configuration section will later be referenced in the `Jenkinsfile`, so be sure to save that name somewhere so you can easily copy/paste it into your pipeline configuration.
+
+![Jenkins Global Tool Configuration](/images/docs/reference/jenkins/global-tool-config.png)
+
+### `Jenkinsfile`
+
+```groovy
+pipeline {
+    agent any
+    
+    stages {
+        stage ("Checkout code") {
+            steps {
+                git url: "git@github.com:your-github-account/you-repo.git",
+                    // Set your credentials id value here.
+                    // See https://jenkins.io/doc/book/using/using-credentials/#adding-new-global-credentials
+                    credentialsId: "yourCredentialsId",
+                    // You could define a new stage that specifically runs for, say, feature/* branches
+                    // and run only "pulumi preview" for those.
+                    branch: "master"
+            }
+        }
+        
+        stage ("Install dependencies") {
+            steps {
+                sh "curl -fsSL https://get.pulumi.com | sh"
+                sh "$HOME/.pulumi/bin/pulumi version"
+            }
+        }
+        
+        stage ("Pulumi up") {
+            steps {
+                // The value "node 8.9.4" is the configuration name in our Global Tool Configuration setup for node.
+                // You should use the name that you used when you added the installation on that page.
+                nodejs(nodeJSInstallationName: "node 8.9.4") {
+                    withEnv(["PATH+PULUMI=$HOME/.pulumi/bin"]) {
+                        sh "cd infrastructure && npm install"
+                        sh "pulumi stack select ${PULUMI_STACK} --cwd infrastructure/"
+                        sh "pulumi up --yes --cwd infrastructure/"
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+## Additional Information
+
+### Single-quotes vs. double-quotes in `Jenkinsfile`
```

Comment:
Note that surrounding `Jenkinsfile` in backticks doesn't look great in the header. cc: @cnunciato

<img width="652" alt="Screen Shot 2019-08-16 at 2 10 55 PM" src="https://user-images.githubusercontent.com/710598/63198486-c43f7600-c02f-11e9-80c6-5f56b00cdcee.png">

Reply:
Yeah doesn't look good. Oh I'll remove the backticks.

Reply:
Yeah, I'll get something up to fix that. 🤦‍♂